### DISCUSSION TITLE
Forwarding of X-Grafana-User header

### DISCUSSION THREAD

**Discussion by svet-b** [grafana/grafana-infinity-datasource#457] [General]
Firstly, thank you for an awesome and hugely versatile plugin! Just trying out some things, and wanted to raise one issue I encountered, which is relevant for us:

In principle, Grafana backend datasource requests should honor the [send_user_header](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#send_user_header) configuration parameter, and forward the `X-Grafana-User` header with any requests that are made. Using the latest Grafana version (v9.2.4) and latest released Infinity version (v1.1.0), I noticed that this header is not passed to requests.

Based on Grafana's own issue tracker https://github.com/grafana/grafana/issues/47734), it seems that this is currently in a somewhat shaky state more generally - e.g. I know it works for some datasources, but not others. It seems that there is some effort to get this straightened out: https://github.com/grafana/grafana/pull/58132, https://github.com/grafana/grafana/pull/58646. I'm curious what the state is with respect to Infinity. E.g. if the Grafana issues are sorted out, will the `X-Grafana-User` header be forwarded automatically? Or does this still require further modifications to the plugin either way?

We're trying to establish an auth strategy for our use case, and knowing whether this will be something we can use (in principle) has some impact on the approach we take.

Thank you again for a wonderful contribution to the community!

**Comment by yesoreyeram** [comment]
Yea. I am closely tracking this. Once merged /release, there is also some amount of work required in plugin side

**Reply by svet-b** [reply]
Thanks for clarifying! A related question (especially in context of https://github.com/grafana/grafana/pull/58132): do you think that cookie forwarding would also be supported? That PR makes it sound like it's a matter of enabling a `keepCookies` switch, but I guess that's something that would need to specifically be set in the plugin (potentially as an option exposed in the UI). If it's as simple as something like that - but not something you'd prioritize in terms of development - then I'd also be quite happy to contribute a PR.

**Comment by yesoreyeram** [comment]
hooo... long time and sorry for the delay. It is finally coming https://github.com/grafana/grafana-infinity-datasource/pull/1072
